src/app/gagestats/gagestats.component.ts

```diff
@@ -46,24 +41,81 @@ export class GagestatsComponent implements OnInit {
     this.loggedInRole = localStorage.getItem('loggedInRole');
     this.title = 'Gage Stats';
     this.timestamp = new Date();
-    this.showStationType = false;
-
     // subscribe to stations subject, which is set in the service's searchStations() function
     this._nssService.Stations.subscribe((s: Array<Station>) => {
         this.selectedStations = s;
-      });
+    });
+    // subscribe to all agencies
+    this._nssService.agencies.subscribe((agencies: Array<Agency>) => {
+      this.agencies = agencies;
+    });
+    // subscribe to all station types
+    this._nssService.stationTypes.subscribe((stationtypes: Array<Agency>) => {
+      this.stationTypes = stationtypes;
+    });
+    //subscribe to page number related information
+    this._nssService.pageResponse.subscribe((pageText: string) => {
+      let response = pageText;
+      this.lastPageNumber = (response.slice(response.indexOf("of") + "of".length));
+      this.lastPageNumber = (this.lastPageNumber.substr(0, this.lastPageNumber.indexOf('.'))); 
+      this.currentPageNumber = (response.slice(response.indexOf("page") + "page".length));
+      this.currentPageNumber = Number(this.currentPageNumber.substr(0, this.currentPageNumber.indexOf('of'))); 
+    });
   }   
 
   showAddStationModal(): void{
     this.gagestatsService.addStation();
   }
 
+  public newPage(event){
+    let pageNumber;
+    if(event.target){
+       pageNumber = (event.target.value);
+    }else{
+       pageNumber = event;
+    }
+    if ((pageNumber - 1) * (pageNumber - this.lastPageNumber) <= 0) {
+      this._nssService.changePageNumber(pageNumber);
+    } else {
+      const toast: Toast = {
+        type: 'warning',
+        title: 'Error',
+        body: 'Page Number Out of Bounds'
+      };
+      this._nssService.showToast(toast);
+    }
+  }
+
+  public getAgencyName(aID) {
+    let acencyName;
+    if (this.agencies) {
+      this.agencies.forEach(z => {
+          if (aID === z.id) {
+            acencyName = z.name;
+          }
+      });
+    }
+    return acencyName;
```

Comment:
This could all be done on one line (and avoid the forEach loop) by using the `.find()` function (same below in getStationType)